src/patch/parse.js

```diff
@@ -7,41 +7,40 @@ export function parsePatch(uniDiff, options = {}) {
     let index = {};
     list.push(index);
 
-    // Ignore any leading junk
+    // Parse diff metadata
     while (i < diffstr.length) {
-      if (/^(Index:|diff -r|@@)/.test(diffstr[i])) {
+      let line = diffstr[i];
+
+      // File header found, end parsing diff metadata
+      if (/^(\-\-\-|\+\+\+|@@)/.test(line)) {
         break;
       }
-      i++;
-    }
-
-    let header = (/^(?:Index:|diff(?: -r \w+)+) (.*)/.exec(diffstr[i]));
-    if (header) {
-      index.index = header[1];
-      i++;
 
-      if (/^===/.test(diffstr[i])) {
-        i++;
+      // Diff index
+      let header = /^(?:Index:|diff(?: -r \w+)+)\s+(.+?)\s*$/.exec(line);
+      if (header) {
+        index.index = header[1];
       }
 
-      parseFileHeader(index);
-      parseFileHeader(index);
-    } else {
-      // Ignore erant header components that might occur at the start of the file
-      parseFileHeader({});
-      parseFileHeader({});
+      i++;
     }
 
+    parseFileHeader(index);
+    parseFileHeader(index);
+
+    // Parse hunks
     index.hunks = [];
 
     while (i < diffstr.length) {
-      if (/^(Index:|diff -r)/.test(diffstr[i])) {
+      let line = diffstr[i];
+
+      if (/^(Index:|diff|\-\-\-|\+\+\+)\s/.test(line)) {
         break;
-      } else if (/^@@/.test(diffstr[i])) {
+      } else if (/^@@\s/.test(line)) {
```

Comment:
I think we should follow the spec and the projects bad using it should fix themselves because being less restrictive here would lead to errors otherwise, but since we already are accepting hunks without file headers (that's not supported by the standard, but there's no technical issues for accept it) and there's other projects using the library that would break, I'll remove the `\s`.